typescript/23-hello-interview-trees/examples.ts

```diff
@@ -0,0 +1,103 @@
+import { TreeNode } from "./trees.ts";
+
+/*
+  Given the root of a binary tree and an integer target,
+  write a recursive function to determine if the tree has
+  a root-to-leaf path where all the values along that path sum to the target.
+*/
+
+export function hasPathSum(root: TreeNode | null, target: number): boolean {
+  if (!root) {
+    return false;
+  }
+
+  if (!root.left && !root.right) {
+    return target === root.val;
+  }
+
+  const left = hasPathSum(root.left, target - root.val);
+  const right = hasPathSum(root.right, target - root.val);
+  return left || right;
+}
+
+/*
+  Given the root of a binary, write a recursive function
+  to determine if it is a valid binary search tree.
+*/
+
+export function validateBST(root: TreeNode | null): boolean {
+  const dfs = (node: TreeNode | null, min: number, max: number): boolean => {
+    if (node === null) {
+      return true;
+    }
+
+    if (node.val <= min || node.val >= max) {
+      return false;
+    }
+
+    return dfs(node.left, min, node.val) && dfs(node.right, node.val, max);
+  };
+
+  return dfs(root, Number.NEGATIVE_INFINITY, Number.POSITIVE_INFINITY);
+}
+
+/*
+  Given the root node of a binary tree, write a recursive function
+  to return the sum of each node's tilt.
+  The tilt of a node is the absolute difference between the sum of its left
+  subtree and the sum of its right subtree. If a node has an empty left or subtree,
+  the sum of the empty subtree is 0.
+ */
+
+export function findTilt(root: TreeNode | null): number {
+  let tilt = 0;
+
+  function dfs(node: TreeNode | null): number {
+    if (!node) return 0;
+
+    const left = dfs(node.left);
+    const right = dfs(node.right);
+
+    tilt += Math.abs(left - right);
+
+    return left + right + node.val;
```

Comment:
It's not immediately obvious what is the return value. Such code needs to be commented.